### PR TITLE
Updated readme based on revised spec & workflow

### PR DESCRIPTION
**Pull request by fershad**
This PR contains a full rewrite of the README for this project. The changes reflect updates made to the specification covering the carbon.txt syntax, how to setup a carbon.txt file, and how to use carbon.txt as a managed service provider.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,148 +1,206 @@
...
+#### For digital service providers with multiple domains
 
-If Netwise ran _all_ their datacentres on renewable power, all they might need to do is list their one provider like so:
+It is to be expected that one digital service provider might own and provide services through multiple domains. In this case, you can maintain a single carbon.txt at one domain (e.g. _<https://my-org.com/.well-known/carbon.txt>_) and refer other domains to that single source of truth. To do this:
 
-```
-Ecotricity Group Ltd
-```
+1. Complete the steps above to register as a provider, and create a carbon.txt file.
+2. **Generate a public key for your account** \
```
Comment (mrchrisadams) [inline README.md]: These hashes are referred to as _domain\_hashes_ in the docs at the link below:

https://greenweb.readthedocs.io/en/latest/how-we-check-for-green.html#

Would you mind updating this to use that language?

File: README.md [hunk 2]
```diff
@@ -1,148 +1,206 @@
...
+   To do this, use the Observable Notebook linked below: \
+   [https://observablehq.com/d/17c922a33a5d6867](https://observablehq.com/d/17c922a33a5d6867) \
+   (Note: We understand this is solution is not ideal & are working on a nicer way to generate secrets)
```
Comment (mrchrisadams) [inline README.md]: Now that the API doc page shows proper responses in production, I really think it's simpler to add an endpoint there to generate a hash from a provided domain, when folks are signed in with the user credentials.

Thank you for making this explicit that this won't be a thing we use for long.

**Review by mrchrisadams — APPROVED** [review]

**Comment by fershad** [comment]
[Settle on and document a specification for carbon.txt](https://trello.com/c/qMH7zaAE/128-settle-on-and-document-a-specification-for-carbontxt)

File: README.md [hunk 3]
```diff
@@ -1,148 +1,211 @@
...
+Providers may want to broadcast that all the websites using their service are also running on green energy. To do this, they can use the [HTTP Via header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/via).
 
-It seems reasonable to expect Krystal to ask Netwise, and ask which datacentre they use for green power, and them to respond with name for it like 'london'. 
+1. Complete the steps above to register as a provider, and create a carbon.txt file.
+2. **Generate a domain hash for your account** \
+   To do this, use the Observable Notebook linked below: \
+   [https://observablehq.com/d/17c922a33a5d6867](https://observablehq.com/d/17c922a33a5d6867) \
+   (Note: We understand this is solution is not ideal & are working on a nicer way to generate secrets)
+3. **Set via headers for on requests generated/processed by your service** \
+   You can now set via headers for requests that go through your service. This allows you to broadcast that the request has come from your service, and point to your carbon.txt file. A valid via header would look like:
```
Comment (jonarnes) [inline README.md]: `requests` should be `response`, I guess, since the service will respond with a via header with reference to the carbon.txt

Comment (fershad) [reply README.md]: Thanks for picking that up @jonarnes

**Pull request merged**